portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,47 +42,50 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         PreparedQuery results = datastore.prepare(query);
 
-        List<Comment> comments = new ArrayList<>();
+        Integer numComments = Integer.parseInt(request.getParameter("num-comments"));
+        CommentObject.Builder commentBuilder;
+
+        List<CommentObject> comments = new ArrayList<>();
         for (Entity entity : results.asIterable()) {
-            long id = entity.getKey().getId();
-            String text = (String) entity.getProperty("text");
-            long timestamp = (long) entity.getProperty("time");
+            if (numComments > 0) {
+                long id = entity.getKey().getId();
+                String text = (String) entity.getProperty("text");
+                long timestamp = (long) entity.getProperty("time");
 
-            Comment comment = new Comment(id, text, timestamp);
-            comments.add(comment);
-        }
+                commentBuilder = CommentObject.newBuilder();
+                commentBuilder.setId(id);
+                commentBuilder.setText(text);
+                commentBuilder.setTime(timestamp);
+                CommentObject hold = commentBuilder.build();
 
-        Gson gson = new Gson();
+                comments.add(hold);
+                numComments--;
+            }
+        }
 
-        String json = "{";
-        json += "\"numComments\": \"" + numComments.toString() + "\", ";
-        json += "\"commentList\":" + gson.toJson(comments) + "}";
+        GsonBuilder gsonBuilder = new GsonBuilder();
+        Gson gson = gsonBuilder.registerTypeAdapter(CommentObject.class, new CommentAdapter()).create();
 
         response.setContentType("application/json;");
-        response.getWriter().println(json);
+        response.getWriter().println(gson.toJson(comments));
     }
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         String comment = request.getParameter("comment-text");
-        String tempNum = request.getParameter("num-comments");
+        System.out.println(comment);
```

Comment:
Don't you want this line and the next to be within the body of 'if (comment.length() > 0)'?

Reply:
No, I just forgot to delete this statement. I added that line in when I discovered that my comments stopped displaying and I was testing to figure out where the problem was. I'll delete the print statement.